aspnetcore/fundamentals/middleware/extensibility-third-party-container/sample/Middleware/IMiddlewareMiddleware.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using MiddlewareExtensibilitySample.Data;
+using MiddlewareExtensibilitySample.Models;
+
+namespace MiddlewareExtensibilitySample.Middleware
+{
+    #region snippet1
+    public class IMiddlewareMiddleware : IMiddleware
```

Comment:
`MiddlewareMiddleware` seems a bit odd. @scottaddie  what do you think? Would `QSlogMiddleware` or something like that be better? I'll go with Scott's vote.

Reply:
That was @davidfowl's instruction. It's the same in the introductory topic. He asked for that and `ConventionalMiddleware`. I kept it the same because presumably the dev will have just come over here from over there, and I wanted it to be as familiar as possible.

Reply:
@Rick-Anderson I agree that `IMiddlewareMiddleware` is a poor name; however, David probably had a good reason for requesting this.

Reply:
It was to distinguish it from the other middleware by how it's activated. It doesn't have to stay this way here, since there's only one middleware in this sample. If it changes here tho, I recommend picking something that can be used in both samples to maintain that consistency in naming across these two topics. After all, it really is exactly the same middleware.

Reply:
Yea, we can change it to something like SimpleInjectorActivatedMiddleware since this is specifically showing how to use another container to activate IMiddleware instances.